wrappers/IWearLogger/src/IWearLogger.cpp

```diff
@@ -78,27 +107,42 @@ class IWearLogger::impl
         return res;
     }
 
-    inline std::string convertSensorNameToValidMatlabVarName(const std::string& sensorName)
+    inline std::string getValidName(std::string& name, const char& c)
     {
-        std::string matlabName{sensorName};
-        // replace special characters with underscore
-        std::replace(matlabName.begin(), matlabName.end(), '#', '_');
-        std::replace(matlabName.begin(), matlabName.end(), '@', '_');
-        std::replace(matlabName.begin(), matlabName.end(), '/', '_');
-        std::replace(matlabName.begin(), matlabName.end(), '(', '_');
-        std::replace(matlabName.begin(), matlabName.end(), ')', '_');
-
-        auto vecStr = split(matlabName, wearable::Separator);
-        std::string validMatlabName;
+        // Replace special characters with desired char c
+        std::replace(name.begin(), name.end(), '#', c);
+        std::replace(name.begin(), name.end(), '@', c);
+        std::replace(name.begin(), name.end(), '/', c);
+        std::replace(name.begin(), name.end(), '(', c);
+        std::replace(name.begin(), name.end(), ')', c);
+
+        auto vecStr = split(name, wearable::Separator);
+
+        std::string validName;
         for (auto& str : vecStr) {
-            if (validMatlabName.empty()) {
-                validMatlabName = str;
+            if (validName.empty()) {
+                validName = str;
             }
             else {
-                validMatlabName = validMatlabName + "_" + str;
+                validName = validName + c + str;
             }
         }
-        return validMatlabName;
+
+        return validName;
+    }
+
+    inline std::string convertSensorNameToValidMatlabVarName(const std::string& sensorName)
+    {
+        std::string matlabName{sensorName};
```

Comment:
This variable is not necessary if the `getValidName` method takes a const reference.

Reply:
Done in https://github.com/robotology/wearables/pull/123/commits/afacfdf1c362f7bf87ef28f853f25037a6a3ca8a